File: template.py
import sys
import os

sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
from lib import lib

INPUT_FILE = os.path.join(os.path.dirname(__file__), 'input.txt')
TEST_FILE = os.path.join(os.path.dirname(__file__), 'test.txt')


def parse_data(content: str):
    '''Parses the data'''
    return content.splitlines()


def silver(content: str, debug: bool = False):
    '''Solves the silver problem'''
    data = parse_data(content)
    return 0


def gold(content: str, debug: bool = False):
    '''Solves the gold problem'''
    data = parse_data(content)
    return 0


def main():
    '''Parses the input and solves the two problems'''
    options = lib.parse_args()

    content = lib.read_file(TEST_FILE)
    print(f"Silver test: {silver(content, debug=True)}")
    print(f"Gold test:   {gold(content, debug=True)}")

    if not options.debug:
        content = lib.read_file(INPUT_FILE)
        print(f"Silver:      {silver(content)}")
        print(f"Gold:        {gold(content)}")


if __name__ == "__main__":
    main()

File: template.js
'use strict';

const minimist = require('minimist');
const fs = require('fs');
const path = require('path');

const INPUT_FILE = path.join(__dirname, 'input.txt');
const TEST_FILE = path.join(__dirname, 'test.txt');

function parseInput(content) {
    return content.split('\n');
}

async function silver(content, test = false) {
    let data = parseInput(content);
    return 42;
}

async function gold(content, test = false) {
    let data = parseInput(content);
    return 42;
}

function parseArgs() {
    return minimist(process.argv.slice(2), {
        boolean: 'debug',
        unknown: (arg) => {
            console.error(`Unknown argument: ${arg}`);
            process.exit(1);
        },
    });
}

async function main() {
    const args = parseArgs();
    let content = fs.readFileSync(TEST_FILE, 'utf-8');

    const [silverTest, goldTest] = await Promise.all([
        silver(content, true),
        gold(content, true)
    ]);

    console.log(`Silver test: ${silverTest}`);
    console.log(`Gold test:   ${goldTest}`);

    if (!args.debug) {
        content = fs.readFileSync(INPUT_FILE, 'utf-8');

        const [silverResult, goldResult] = await Promise.all([
            silver(content),
            gold(content)
        ]);

        console.log(`Silver:      ${silverResult}`);
        console.log(`Gold:        ${goldResult}`);
    }
}

main();
